src/app/store/chat/chat.service.ts

```diff
@@ -130,17 +130,21 @@ export class ChatService {
 
   // simulating an api call that returns created message with real id and time
   // to-do: connect to actual api
-  sendMessage(message: { userId: string; body: string }): Observable<Message> {
+  sendMessage(message: {
+    id: string;
+    userId: string;
+    body: string;
+  }): Observable<Message> {
     const response = (): Message => ({
       body: message.body,
       userId: message.userId,
-      id: crypto.randomUUID(),
+      id: message.id || crypto.randomUUID(),
```

Comment:
Re-use ID for update case.

Reply:
I didn't do it this way because even if unlikely to generate conflicting IDs, we shouldn't rely on the front end to generate IDs

Reply:
the ID generated by the backend might be used later for features like updating or deleting messages

Reply:
Just separate the backend ID and the frontend one. Frontend ID should persist.